docs/src/DeveloperDocumentation/styleguide.md

```diff
@@ -241,3 +217,213 @@ existing block.
+- variable names/generators/relations should not be printed only their number.
+- Only the first word is capitalized e.g. `Polynomial ring`
+- one should use `:supercompact` for nested printing in compact
+- nested calls to `one line` (if you think them really necessary) should be at the end,
+  so that one can read sequentially. Calls to `:supercompact` can be anywhere.
+- commas must be enclosed in brackets so that printing tuples stays unambiguous
+#### Super compact printing
+- a user readable version of the main (mathematical) type.
+- a single term or a symbol/letter mimicking mathematical notation
+- should usually only depend on the type and not of the type parameters or of
+  the concrete instance - exceptions of this rule are possible e.g. for `GF(2)`
+- no nested printing. In particular variable names and `base_ring` must not be displayed.
+  This ensures that `one line` and `:supercompact` stay compact even for complicated things.
+  If you want nested printing use `one line` or `detailed`.
+
+### Implementing show functions
+
+Here is the translation between `:detail`, `one line` and `:supercompact`.
+
+```
+print(io, "text/plain", x)                 # detailed printing
+print(io, x)                               # one line printing
+print(IOContext(:supercompact => true), x) # supercompact printing
+```
+
+For reference, string interpolation `"$(x)"` will also use `print(io, x)`.
+
+#### Mockup
+
+##### Detailed printing with a new line.
+
+```julia
+struct NewRing
+  base_ring
+end
+```
+This implements the detailed printing. Note that the new line is needed for
+this to work. The last print statement must not add a new line.
+
+```julia
+function Base.show(io::IO, ::MIME"text/plain", R::NewRing)
+  println(io, "I am a new ring")
+  println(io, "I print with newlines")
+  print(io, R.base_ring)
+end
+```
+
+This takes care of one line and `:supercompact` printing
+```julia
+function Base.show(io::IO, R::NewRing)
+  if get(io, :supercompact, false)
+    # no nested printing
+    print(io, "supercompact printing of newring ")
+  else
+    # nested printing allowed, preferably supercompact
+    print(io, "one line printing of newring with ")
+    print(IOContext(io, :supercompact => true), "supercompact ", R.base_ring)
+  end
+end
+```
+
+
+##### Detailed printing in a single line.
+
+This version needs to be used in case the detailed
+printing does not contain newlines.
+Then detailed and one line printing agree.
+The if clause takes care of supercompact printing as well.
+```julia
+function Base.show(io::IO, R::NewRing)
+  if get(io, :supercompact, false)
+    # no nested printing
+    print(io, "supercompact printing of newring")
+  else
+    # nested printing allowed, preferably supercompact
+    println(io, "I am a new ring and always print in one line" )
+    print(IOContext(io, :supercompact => true), base_ring(R))
+  end
+end
+```
+
+##### The following is not working as expected and should not be used
```

Comment:
It is just how julia works. Nothing on the Oscar side here.

Reply:
OK, but I still find this confusing when looked at in isolation (i.e. when one gets here via a link and not via reading the whole section sequentially). So I would just add a little bit extra (repeating a few things that were said before)
```suggestion
##### The following is not working as expected and should not be used

This example does not work correctly because the `detailed` printing does not
include a newline, which is expected by the Julia printing system. To correctly
support single line `detailed` printing, read the preceding section.
```